FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,58 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				do {
+					let items = try FeedImageMapper.map(data, response)
+					completion(.success(items))
+				} catch {
+					completion(.failure(Error.invalidData))
+				}
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			}
+		})
+	}
+}
+
+private class FeedImageMapper {
+	private struct Root: Decodable {
+		let items: [Item]
+	}
+
+	private struct Item: Hashable, Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let url: URL
```

Comment:
The `Item` properties can be named exactly as the Feed API specs/contract. This way, the code is shorter and more concise since it doesn't need the `CodingKeys` enum.

Using underscores is not the convention in Swift, but in this case, this struct exists to describe remote models with different conventions. So it is ok that the remote model's properties don't follow swift's convention - it follows the remote model's convention!

Following the remote model's convention here clarifies the intent of the code and makes maintenance easier (especially when you have many remote models to represent in the app!)
```suggestion
		let image_id: UUID
		let image_desc: String?
		let image_loc: String?
		let image_url: URL
```